### PR TITLE
Juliet

### PR DESCRIPTION
**Pull request by Na-Cly**
## Description

Added juliet keyboard. Pro micro version of the ROMEO. I don't expect it to be widely available so I added it to the handwired folder. PCB files are available, but as of now only 5 exist. 

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/handwired/juliet/config.h [hunk 1]
```diff
@@ -0,0 +1,247 @@
...
+#define LCD_DATA2_PIN    6            //< pin for 4bit data bit 2
+#define LCD_DATA3_PIN    7            //< pin for 4bit data bit 3
+#define LCD_RS_PORT      LCD_PORT     //< port for RS line
+#define LCD_RS_PIN       3            //< pin  for RS line
+#define LCD_RW_PORT      LCD_PORT     //< port for RW line
+#define LCD_RW_PIN       2            //< pin  for RW line
+#define LCD_E_PORT       LCD_PORT     //< port for Enable line
+#define LCD_E_PIN        1            //< pin  for Enable line
+#endif
+*/
```
Comment (fauxpark) [inline keyboards/handwired/juliet/config.h:242]:
```suggestion
```

File: keyboards/handwired/juliet/rules.mk [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+#   LUFA DFU     lufa-dfu
+#   QMK DFU      qmk-dfu
+#   ATmega32A    bootloadHID
+#   ATmega328P   USBasp
+BOOTLOADER = caterina
+
+# disable debug code
+OPT_DEFS = -DDEBUG_LEVEL=0
+
+
```
Comment (fauxpark) [inline keyboards/handwired/juliet/rules.mk:17]:
```suggestion
```

File: keyboards/handwired/juliet/rules.mk [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
+NKRO_ENABLE = no            # USB Nkey Rollover
+BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
+RGBLIGHT_ENABLE = no        # Enable keyboard RGB underglow
+MIDI_ENABLE = no            # MIDI support
+UNICODE_ENABLE = no         # Unicode
+BLUETOOTH_ENABLE = no       # Enable Bluetooth with the Adafruit EZ-Key HID
+AUDIO_ENABLE = no           # Audio output on port C6
+FAUXCLICKY_ENABLE = no      # Use buzzer to emulate clicky switches
+HD44780_ENABLE = no         # Enable support for HD44780 based LCDs
```
Comment (fauxpark) [inline keyboards/handwired/juliet/rules.mk:37]:
```suggestion
```

**Comment by Na-Cly** [comment]
> In its current form, this PR cannot be accepted due to the submodule changes in `lib/chibios`, `lib/chibios-contrib`, `lib/ugfx`.

What’s the way to correct this? Did I try to merge older files? I didn’t intentionally modify those files.

**Comment by noroadsleft** [comment]
> > In its current form, this PR cannot be accepted due to the submodule changes in `lib/chibios`, `lib/chibios-contrib`, `lib/ugfx`.
> 
> What’s the way to correct this? Did I try to merge older files? I didn’t intentionally modify those files.

image

You've based this work upon an outdated version of QMK Firmware (`1596 commits behind qmk:master`), and some of our dependencies have been updated and reworked. You'll need to [rebase](https://git-scm.com/docs/git-rebase) this on a current QMK version.

**Comment by Na-Cly** [comment]
Ok I think I figured it out. Please let me know if the changes didn't reflect correctly here. 

Thanks guys!

**Comment by Na-Cly** [comment]
Got it. Should be good now. Let me know if you notice anything else.

Thanks again guys.

**Pull request merged**